polkadot/node/network/collator-protocol/src/collator_side/metrics.rs

```diff
@@ -428,11 +579,22 @@ impl CollationStats {
 
 	/// Returns the age of the collation at the moment of inclusion.
 	pub fn included(&self) -> Option<BlockNumber> {
-		let included_at = self.included_at?;
+		let included_at = self.included_at.first().map(|(block_number, _)| *block_number)?;
 		let backed_at = self.backed_at?;
 		Some(included_at.saturating_sub(backed_at))
 	}
 
+	/// Returns true if the collation is included at the given block number and hash.
+	pub fn is_included_at(&self, block_number: BlockNumber, hash: H256) -> bool {
+		self.included_at.iter().any(|(b, h)| *b == block_number && *h == hash)
+	}
+
+	/// Returns the age of the collation at the moment of finalization.
+	pub fn finalized(&self) -> Option<BlockNumber> {
+		let finalized_at = self.finalized_at?;
+		Some(finalized_at.saturating_sub(self.relay_parent_number))
```

Comment:
With building on older relay chains this becomes a bit misleading, because the collation wasn't built at self.relay_parent_number, it was built later.